### PR TITLE
feat: mysterybox whitelist check

### PR DESCRIPTION
**Pull request by Randolph314**
## Description

Add whitelist function to maskbox

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [x] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/mask/src/plugins/MaskBox/apis/merkleProof.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+            urlcat(MERKLE_PROOF_ENDPOINT, {
+                leaf,
+                root: root.replace(/^0x/, ''),
+            }),
+        )
+        return (await response.json()) as { proof?: string[]; message?: string; module?: string }
+    } catch (err: any) {
+        throw new Error(err?.message)
+    }
+}
```
Comment (guanbinrui) [inline packages/mask/src/plugins/MaskBox/apis/merkleProof.ts:16]:
```suggestion
    } catch (error: unknown) {
        throw error
    }
}
```

File: packages/mask/src/plugins/MaskBox/hooks/useContext.ts [hunk 2]
```diff
@@ -133,10 +145,38 @@ function useContext(initialState?: { boxId: string }) {
...
+    const qualification = useQualification(
+        boxInfo?.qualificationAddress,
+        account,
+        value?.proof ? coder.encodeParameters(['bytes', 'bytes32'], [proofBytes, rootHash]) : undefined,
+    )
+
+    // not in whitelist
+    const notInWhiteList = value?.message === 'leaf not found'
+
+    console.log('qualification', qualification?.qualified)
```
Comment (guanbinrui) [inline packages/mask/src/plugins/MaskBox/hooks/useContext.ts]:
```suggestion
```

File: packages/mask/src/plugins/MaskBox/apis/merkleProof.ts (hunk 1 above)
Comment (Jack-Works) [reply packages/mask/src/plugins/MaskBox/apis/merkleProof.ts:16]: then why you need to add a try-catch anyway?

File: packages/mask/src/plugins/MaskBox/hooks/useContext.ts (hunk 2 above)
Comment (Randolph314) [reply packages/mask/src/plugins/MaskBox/hooks/useContext.ts]: done

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**